adapter/internal/adapter/adapter.go

```diff
@@ -255,11 +257,16 @@ func Run(conf *config.Config) {
 
 		if isAzureEventingFeatureFlagEnabled {
 			logger.LoggerMgw.Info("[TEST][FEATURE_FLAG_REPLACE_EVENT_HUB] Starting to integrate with azure service bus")
-			messaging.InitiateAndProcessEvents(conf)
+			var connectionURLList = conf.ControlPlane.BrokerConnectionParameters.EventListeningEndpoints
+			if strings.Contains(connectionURLList[0], amqpProtocol) {
+				go messaging.ProcessEvents(conf)
+			} else {
+				messaging.InitiateAndProcessEvents(conf)
```

Comment:
For the eventhub its done in go routine, for ASB its not, don't we have to be consistant

Reply:
InitiateAndProcessEvents will initiate connections and meta data and if any error it will avoid adapter from starting , inside InitiateAndProcessEvents after initiation is done other stuff are running relevant go routine. I do not think we need to run this also as a go routine

Reply:
also ProcessEvents will anyway going to be depricated

Reply:
The reason why we didn't want to have the go routine was to block the server startup if there is any issue with during connection initiation. But with the exsisting broker in a current deployment rollout, there'll be an almost definite round of reconnection attempts and AFAIR @Rajith90 mentioned this was changed to have a go routine because the server started hanging. 

Therefore I believe we can leave this as it it?